### PR TITLE
Translations

### PR DESCRIPTION
**Pull request by simonewebdesign**

### CODE REVIEW

**Comment by danmarcab** [comment]
Looks good to me

File: lib/translations.ex [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    generate([], path)
+  end
+
+  @doc """
+  Generates Translation.elm according to the chosen languages and the path to
+  gettext folder.
+  """
+  def generate(languages, path) do
+    available_languages =
+      [ {"it_IT", {"Italian", "Italiano"}},
```
Comment (yelps) [inline lib/translations.ex]: Why are the languages still hard coded?

**Comment by law326** [comment]
👍

File: lib/translations.ex (hunk 1 above)
Comment (simonewebdesign) [inline lib/translations.ex]: We can take care of this in PX-612, for now I'd rather merge this to unblock the other repositories.

File: lib/template.eex [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+# Ports
+@docs saveLocale
+-}
+
+import Dict
+
+{-| Currently supported languages.
+-}
+type Language
+    = English<%= for {_, {lang, _}} <- languages do %>
```
Comment (yelps) [inline lib/template.eex]: We should make English configurable as well in the next step (PX-612).

File: lib/template.eex [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+fromLocaleLanguage str =
+    case str of
+        "en" ->
+            English
+
+        <%= for {locale, {lang, _}} <- languages do %>"<%= String.slice locale, 0, 2 %>" ->
+            <%= lang %>
+
+        <% end %>default ->
+            English
```
Comment (yelps) [inline lib/template.eex]: We should also make the default language configurable in the next step.

File: lib/template.eex (hunk 2 above)
Comment (simonewebdesign) [inline lib/template.eex]: Will do!

File: lib/translations.ex (hunk 1 above)
Comment (yelps) [inline lib/translations.ex]: 👍

File: lib/template.eex (hunk 2 above)
Comment (yelps) [inline lib/template.eex]: 👍

File: lib/translations.ex [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+  @doc """
+  Displays a help message.
+  """
+  def process(:help) do
+    IO.puts @moduledoc
+    System.halt(0)
+  end
+
+  def process(%{languages: languages, path: path}) do
```
Comment (mtanzi) [inline lib/translations.ex]: I think it's easier to read if you split it in 2 functions checking the condition there:

```elixir
def process(%{languages: languages, path: ""}) do
  generate(languages, ".")
end

def process(%{languages: languages, path: path}) do
  generate(languages, path)
end
```

File: lib/translations.ex [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    generate(languages, path)
+  end
+
+  def process(%{languages: languages}) do
+    path = "."
+    generate(languages, path)
+  end
+
+  def process(%{path: path}) do
+    path = if path == "", do: ".", else: path
```
Comment (mtanzi) [inline lib/translations.ex]: Same as above

**Comment by yelps** [comment]
Besides some comments for the next steps it looks great! Well done! 👍

**Comment by simonewebdesign** [comment]
Merging so that we unblock these:
- https://github.com/MyMedsAndMe/px_data_capture_ui/pull/434
- https://github.com/MyMedsAndMe/px_pharma_ui/pull/12

File: lib/translations.ex [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    generate([], path)
+  end
+
+  @doc """
+  Generates Translation.elm according to the chosen languages and the path to
+  gettext folder.
+  """
+  def generate(languages, path) do
```
Comment (mtanzi) [inline lib/translations.ex]: I would try to break down the `generate` function. I think it does to many things now

**Comment by mtanzi** [comment]
a couple of comments, otherwise looks good

**Comment by simonewebdesign** [comment]
@mtanzi I'll refactor with your suggestions in the next step and ping you when it's done. Merging now!

**Pull request merged**